libs/cpp-ledger/ledger/device_hid.cpp

```diff
@@ -0,0 +1,173 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "cpp-ledger/ledger/device_hid.hpp"
+
+#include "cpp-ledger/ledger/const.hpp"
+#include "cpp-ledger/ledger/utils.hpp"
+
+namespace ledger {
+
+  DeviceHid::DeviceHid() {
+    info.path = nullptr;
+    info.serial_number = nullptr;
+    info.manufacturer_string = nullptr;
+    info.product_string = nullptr;
+    info.next = nullptr;
+  }
+
+  DeviceHid::~DeviceHid() {
+    hid_close(device);
+
+    if (info.path != nullptr) {
+      delete[] info.path;
+    }
+
+    if (info.serial_number != nullptr) {
+      delete[] info.serial_number;
+    }
+
+    if (info.manufacturer_string != nullptr) {
+      delete[] info.manufacturer_string;
+    }
+
+    if (info.product_string != nullptr) {
+      delete[] info.product_string;
+    }
+  }
+
+  DeviceHid::DeviceHid(DeviceHid &&other) {
```

Comment:
move constructor should be marked noexcept
```suggestion
  DeviceHid::DeviceHid(DeviceHid &&other) noexcept {
```